pkg/api/norman/customization/gke/handler.go

```diff
@@ -0,0 +1,194 @@
+			logrus.Debugf("[gke-handler] error getting machine types: %v", err)
+			handleErr(writer, errCode, err)
+			return
+		}
+		writer.Write(serialized)
+	case "gkeNetworks":
+		if serialized, errCode, err = listNetworks(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting networks: %v", err)
+			handleErr(writer, errCode, err)
+			return
+		}
+		writer.Write(serialized)
+	case "gkeServiceAccounts":
+		if serialized, errCode, err = listServiceAccounts(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting serviceaccounts: %v", err)
+			handleErr(writer, errCode, err)
+			return
+		}
+		writer.Write(serialized)
+	case "gkeSubnetworks":
+		if serialized, errCode, err = listSubnetworks(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting subnetworks: %v", err)
+			handleErr(writer, errCode, err)
+			return
+		}
+		writer.Write(serialized)
+	case "gkeVersions":
+		if serialized, errCode, err = listVersions(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting versions: %v", err)
+			handleErr(writer, errCode, err)
+			return
+		}
+		writer.Write(serialized)
+	case "gkeZones":
+		if serialized, errCode, err = listZones(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting zones: %v", err)
+			handleErr(writer, errCode, err)
+			return
+
+		}
+		writer.Write(serialized)
+	default:
+		handleErr(writer, httperror.NotFound.Status, fmt.Errorf("invalid endpoint %v", resourceType))
+	}
+}
+
+func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credID string) (int, error) {
+	ns, name := ref.Parse(credID)
+	if ns == "" || name == "" {
+		logrus.Debugf("[GKE] invalid cloud credential ID %s", credID)
+		return httperror.InvalidBodyContent.Status, fmt.Errorf("invalid cloud credential ID %s", credID)
+	}
+
+	var accessCred client.CloudCredential //var to check access
+	if err := access.ByID(h.generateAPIContext(req), &schema.Version, client.CloudCredentialType, credID, &accessCred); err != nil {
+		if apiError, ok := err.(*httperror.APIError); ok {
+			if apiError.Code.Status == httperror.PermissionDenied.Status || apiError.Code.Status == httperror.NotFound.Status {
+				return httperror.InvalidBodyContent.Status, fmt.Errorf("cloud credential not found")
+			}
+		}
+		return httperror.NotFound.Status, err
+	}
+
+	cc, err := h.secretsLister.Get(ns, name)
+	if err != nil {
+		logrus.Debugf("[GKE] error accessing cloud credential %s", credID)
+		return httperror.InvalidBodyContent.Status, fmt.Errorf("error accessing cloud credential %s", credID)
+	}
+	cap.Credentials = string(cc.Data["googlecredentialConfig-authEncodedJson"])
+
+	cap.ProjectID = req.URL.Query().Get("projectId")
+	if cap.ProjectID == "" {
+		logrus.Debugf("[GKE] error getting projectId")
+		return httperror.InvalidBodyContent.Status, fmt.Errorf("error getting projectId")
```

Comment:
Why `InvalidBodyContent` rather than `BadRequest`? It's inconsistent with the other validation https://github.com/rancher/rancher/pull/31759/files#diff-3791cf107be82e12b9080e5e2ad658a28e794a29522d2a446f9d833c571102daR173

Reply:
I have changed it to `http.StatusBadRequest`